### ISSUE TITLE
020_Skins reference GLTF file or script

### ISSUE THREAD

**Issue by daylanKifky** [KhronosGroup/glTF-Tutorials#93]
I'm trying to replicate the basic example found in `020_Skins`. I created a script using pygltflib that implements the steps from that chapter of the tutorial as precisely as possible. However, I'm encountering small errors and inconsistencies in the skeleton morphology. In particular:

- The gLTF validator gives me this error:
`ACCESSOR_INVALID_IBM	Matrix element at index 23 (component index 7) contains invalid value: -1.	/skins/0/inverseBindMatrices`

- The bind pose of the skeleton has both nodes at the origin, they assume the correct position when posed, but the mesh ends up with an incorrect transformation (stretched vertically)

![image](https://github.com/user-attachments/assets/0501f604-d774-4633-aa4e-d9cc51744396)

This is probably not an specific error with the tutorial, but any other information or clarification will be greatly appreciated

Here's the script that generates the GLTF:
https://pastebin.com/ZXyFSr8Q

**Comment by javagl** [comment]
The skinning again 🙂 This section underwent a few iterations, because I struggled to get it "right" and in a way that could help others to understand how it works. The last iteration was done in https://github.com/KhronosGroup/glTF-Tutorials/pull/64 , which contains many valuable explanations by emackey and scurest. It may be a bit lengthy, but might be work a look.

**But....** it _might_ not be necessary to (further) dive into the details here. From quickly looking at the Python script, the data from the IBM accessor of the original model in the tutorial, and the model that is generated by the Python script....

![SimpleSkin Tutorial Issue](https://github.com/user-attachments/assets/5f6d24cc-12a4-4bf5-91a3-6f9b078b4b1d)

... one can see that it is only a "row-major-vs-column-major" issue. Changing the second IBM to
```
    # Second joint inverse bind matrix (translate back by -1.0 in Y)
    [
        1.0, 0.0, 0.0, 0.0,
        0.0, 1.0, 0.0, 0.0,
        0.0, 0.0, 1.0, 0.0,
        0.0, -1.0, 0.0, 1.0
    ]
```
should fix it. (There are no animations yet, but ... that's unrelated, I guess...)

**Comment by daylanKifky** [comment]
Many thanks for your quick response! It sounds like it's probably that. I couldn't wrap my mind around why a -1 would be an invalid value. Will test it asap

**Comment by daylanKifky** [comment]
It worked :)
![image](https://github.com/user-attachments/assets/560ba214-d603-4144-8b07-4f42f5736a04)

I see #64 is already merged to `main`, to the tutorial I was looking at probably already contained those mods, or am I missing something?

**Comment by javagl** [comment]
> the tutorial I was looking at probably already contained those mods

Yes, the tutorial contains the latest state, including these updates. 

The PR only provides some background and context about _why_ certain values (like the entries of the inverse bind matrices) are the way they are.

**Comment by daylanKifky** [comment]
great, thanks again!
